core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java

```diff
@@ -112,11 +117,15 @@
         this.port = requireNonNull(port, "port");
         this.sslContexts = sslContexts;
         this.gracefulShutdownSupport = requireNonNull(gracefulShutdownSupport, "gracefulShutdownSupport");
+        maxConnectionAgeMillis = config.maxConnectionAgeMillis();
     }
 
     @Override
     protected void initChannel(Channel ch) throws Exception {
         ChannelUtil.disableWriterBufferWatermark(ch);
+        if (maxConnectionAgeMillis > 0) {
+            ch.attr(CONNECTION_START_TIME_NANO).set(System.nanoTime());
```

Comment:
That makes sen. I think we can record the time in `ServerHttp*ObjectEncoder.<init>` and store it as a final member field. Because if a connection does not send packets, `KeepAliveHandler` will close the connection.